### PR TITLE
Pretty error if loading malformed js/json

### LINKED ISSUE #186: Server barfs when confronted with invalid JSON table file
```
C:\Users\ADRIANHA\Documents\GitHub\testsite [mobileapps-starter +1 ~0 -0 !]> node --debug server.js
Debugger listening on port 5858
In azureMobile.js...
2015-11-12T21:50:11.319Z - verbose: Using SQL Server data source, server: ahall-mobileapps-starter-sql.database.windows.net:default, user: azureadmin@ahall-mobileapps-starter-sql
2015-11-12T21:50:11.319Z - verbose: SQL Azure database detected - setting connection encryption
2015-11-12T21:50:11.319Z - verbose: Using SQL Server data source, server: ahall-mobileapps-starter-sql.database.windows.net:default, user: azureadmin@ahall-mobileapps-starter-sql
2015-11-12T21:50:11.319Z - verbose: SQL Azure database detected - setting connection encryption
module.js:463
    throw err;
    ^

SyntaxError: C:\Users\ADRIANHA\Documents\GitHub\testsite\tables\TaskItem.json: Unexpected string
    at Object.parse (native)
    at Object.Module._extensions..json (module.js:460:27)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at loadModule (C:\Users\ADRIANHA\Documents\GitHub\testsite\node_modules\azure-mobile-apps\src\configuration\loader.js:39:24)
    at C:\Users\ADRIANHA\Documents\GitHub\testsite\node_modules\azure-mobile-apps\src\configuration\loader.js:56:13
    at Array.forEach (native)
    at loadDirectory (C:\Users\ADRIANHA\Documents\GitHub\testsite\node_modules\azure-mobile-apps\src\configuration\loader.js:51:32)
C:\Users\ADRIANHA\Documents\GitHub\testsite [mobileapps-starter +1 ~1 -0 !]>
```

### PR DESCRIPTION
**Pull request by mamaso**
resolves #186

### CODE REVIEW

File: src/configuration/loader.js [hunk 1]
```diff
@@ -36,8 +36,17 @@ module.exports = {
...
-        loadedModule = require(targetPath),
-        targetModule = target[moduleName] || {};
+        targetModule = target[moduleName] || {},
+        loadedModule;
+
+    try {
+        logger.verbose('Attempting to load ' + targetPath);
+        loadedModule = require(targetPath);
+    } catch (err) {
+        var error = targetPath + ' is formatted incorrectly';
```
Comment (danderson00) [inline src/configuration/loader.js]: I prefer an error message of something like "Unable to load " + targetPath. Other errors such as filesystem permissions could cause an exception to be thrown - the exception message should contain enough detail.

Comment (mamaso) [inline src/configuration/loader.js]: fair point

File: src/configuration/loader.js [hunk 2]
```diff
@@ -36,8 +36,17 @@ module.exports = {
 
 function loadModule(target, targetPath) {
     var moduleName = path.basename(targetPath, path.extname(targetPath)),
-        loadedModule = require(targetPath),
-        targetModule = target[moduleName] || {};
+        targetModule = target[moduleName] || {},
+        loadedModule;
+
+    try {
+        logger.verbose('Attempting to load ' + targetPath);
```
Comment (danderson00) [inline src/configuration/loader.js]: Bump this log level up to either debug or silly

File: src/configuration/loader.js [hunk 3]
```diff
@@ -36,8 +36,17 @@ module.exports = {
...
+        loadedModule = require(targetPath);
+    } catch (err) {
+        var error = 'Unable to load ' + targetPath;
+        logger.error(error);
+        throw new Error(error);
```
Comment (danderson00) [inline src/configuration/loader.js]: So this is interesting... Do we log the exception here, or rely on consumers to log? I think we both log and throw like you have here, but do the following:

    logger.error('Unable to load ' + targetPath, err); // include the actual error in the log statement
    throw err; // don't wrap the error in a new error

Comment (mamaso) [inline src/configuration/loader.js]: I like it

**Pull request merged**